src/casync.c

```diff
@@ -834,7 +835,7 @@ static int ca_sync_process_decoder_request(CaSync *s) {
                 if (r < 0)
                         return r;
 
-                if (index_size != object_size) {
+                if ((size_t) index_size != object_size) {
```

Comment:
Well, I got a warning, otherwise I wouldn't have patched it:
```
src/casync.c: In function ‘ca_sync_process_decoder_request’:
src/casync.c:821:57: warning: passing argument 3 of ‘ca_index_read_object’ from incompatible pointer type [-Wincompatible-pointer-types]
                 r = ca_index_read_object(s->index, &id, &index_size);
                                                         ^
In file included from src/casync.c:8:0:
src/caindex.h:21:5: note: expected ‘uint64_t * {aka long long unsigned int *}’ but argument is of type ‘size_t * {aka unsigned int *}’
 int ca_index_read_object(CaIndex *i, CaObjectID *id, uint64_t *ret_size);
     ^
```